asu/build.py

```diff
@@ -202,7 +207,7 @@ def build(build_request: BuildRequest, job=None):
         )
         log.debug(f"Diffed packages: {build_cmd_packages}")
 
-    job.meta["imagebuilder_status"] = "calculate_packages_hash"
+    job.meta["imagebuilder_status"] = "validate_manifest"
```

Comment:
I'm fine to change this but it needs to change in the Luci App too

Reply:
I've added a couple other new status "types", so I'll add all of them to LuCI app.
```
asu/build.py:    job.meta["imagebuilder_status"] = "init"
asu/build.py:    job.meta["imagebuilder_status"] = "container_setup"
asu/build.py:    job.meta["imagebuilder_status"] = "validate_revision"
asu/build.py:    job.meta["imagebuilder_status"] = "validate_manifest"
asu/build.py:    job.meta["imagebuilder_status"] = "calculate_packages_hash"
asu/build.py:    job.meta["imagebuilder_status"] = "building_image"
asu/build.py:    job.meta["imagebuilder_status"] = "done"
asu/util.py:    job.meta["imagebuilder_status"] = "failed"
```

Reply:
LuCI app looks like it's got some bitrot, it includes obsolete steps and is missing others.  How should we handle this?  I'm tempted to keep all the old ones (like `download_imagebuilder`) for backward compatibility, and add all the new ones in the proper order.

Reply:
Also, I've got timing data on the build steps.  Should we tune the "percent done" values to be more realistic, or just leave them?  The big problem is that the image pull and make image steps are the big one, and can vary wildly.  Like this ath79 build, where image pull takes 4x longer than the actual build...

```
asu_worker_1 15:16:00 Timing analysis for container-archiving
asu_worker_1     0    0.000    0.000   0.0%  -start-
asu_worker_1     1    0.090    0.090  80.7%  archive-get /builder/6d9c70345bf31eb7d5db7f1a98397829
asu_worker_1     2    0.101    0.011  10.1%  archive-write /tmp/tmpvq6g8wgb.tar
asu_worker_1     3    0.111    0.010   9.1%  archive-extract /tmp/tmpvq6g8wgb.tar: 7 members, 8_145_846 bytes
asu_worker_1     4    0.111    0.000   0.0%  -stop-
asu_worker_1 total    0.111    0.111 100.0%
asu_worker_1 15:16:00 Signing images: ['openwrt-e0a0070be4da-ath79-generic-tplink_archer-c7-v4-squashfs-factory.bin', 'openwrt-e0a0070be4da-ath79-generic-tplink_archer-c7-v4-squashfs-sysupgrade.bin']
asu_worker_1 15:16:00 No build key found, skipping signing
asu_worker_1 15:16:00 Timing analysis for build-SNAPSHOT-ath79/generic (15 packages)
asu_worker_1     0    0.000    0.000   0.0%  -start-
asu_worker_1     1    0.002    0.002   0.0%  store-cleanup
asu_worker_1     2    0.525    0.524   0.9%  image-configuration
asu_worker_1     3   28.462   27.936  50.4%  image-pull
asu_worker_1     4   33.145    4.683   8.5%  container-create
asu_worker_1     5   33.362    0.217   0.4%  make-info
asu_worker_1     6   33.363    0.001   0.0%  make-manifest-setup
asu_worker_1     7   48.507   15.145  27.3%  make-manifest
asu_worker_1     8   48.508    0.001   0.0%  make-image-setup
asu_worker_1     9   55.362    6.854  12.4%  make-image (includes container-archiving)
asu_worker_1    10   55.415    0.053   0.1%  image-validation
asu_worker_1    11   55.415    0.000   0.0%  image-signing
asu_worker_1    12   55.419    0.004   0.0%  build-response
asu_worker_1    13   55.419    0.000   0.0%  -stop-
asu_worker_1 total   55.419   55.419 100.0%
asu_worker_1 15:16:00 default: Job OK (6d9c70345bf31eb7d5db7f1a98397829)
asu_worker_1 15:16:00 Result is kept for 600 seconds

```

Reply:
LuCI app done. https://github.com/openwrt/luci/pull/7329